algebras/algebra/src/main/scala/endpoints4s/algebra/Responses.scala

```diff
@@ -144,6 +144,14 @@ trait Responses extends StatusCodes with InvariantFunctorSyntax {
       headers: ResponseHeaders[B] = emptyResponseHeaders
   )(implicit tupler: Tupler.Aux[A, B, R]): Response[R]
 
+  /** Add the provided `headers` to the `response`.
+    */
+  def addResponseHeaders[A, H](
+      response: Response[A],
+      headers: ResponseHeaders[H]
+  )(implicit tupler: Tupler[A, H]): Response[tupler.Out] =
```

Comment:
I just noticed that there is a small issue with this operation. It adds a header to a `Response`. However a response can be a list of possible responses (when you use `orElse` between responses). In such a case, `addResponseHeaders` adds the header to all the possible responses. Not sure whether this is a problem or not.

Reply:
I am not sure either, I am also not entirely sure when you would use this function. It is also relatively easy to workaround by using the `addResponseHeaders` before the `orElse` operation. Might be more of a caveat that needs to be documented (inline) than something that needs to be fixed. WDYT?

Reply:
True, this is not really a problem. I just found it surprising when I realized it. I am fine with keeping the method as it is.